server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java

```diff
@@ -89,6 +111,24 @@ private void addServices(ServerBuilder sb) {
         new IcebergRestCatalogService(catalogService, schemaService, tableService, metadataService),
         icebergRequestConverter,
         icebergResponseConverter);
+
+    ServerPropertiesUtils serverPropertiesUtils = ServerPropertiesUtils.getInstance();
+    String authorization = serverPropertiesUtils.getProperty("server.authorization");
+    // TODO: eventually might want to make this secure-by-default.
+    if (authorization != null && authorization.equalsIgnoreCase("enable")) {
+      LOGGER.info("Authorization enabled.");
+      AuthDecorator authDecorator = new AuthDecorator();
+      ExceptionHandlingDecorator exceptionDecorator = new ExceptionHandlingDecorator(new GlobalExceptionHandler());
+      sb.decorator(basePath + "catalogs", authDecorator)
```

Comment:
is there a way to do this so all paths by default have the auth decorator applied and we can just add exemptions (the token endpoint, etc.)?

Reply:
I'll look again. I agree it would be nice to be "secure by default" with exceptions.

Reply:
Ok was able to figure out how to secure all endpoints with exceptions.